### ISSUE TITLE
How will the new League of Legends anti cheat react to Pengu's injection into the LCU?

### ISSUE THREAD

**Issue by Soraclee** [PenguLoader/PenguLoader#108]
I wanted to know if you've thought of a solution to this because people could get bans super easily when they just want to use the league of legends client api, it seems to me that pengu uses dll injection to put it in the client. 

Riot Article : https://support-leagueoflegends.riotgames.com/hc/en-us/articles/24169857932435-Riot-Vanguard-League-of-Legends-

That's what I wanted to know.

**Comment by BakaFT** [comment]
Riot should not ban external API calls harshly.

But for Pengu things become complicated because it's internal somehow. as you said we actutally inject `core.dll` into the client, which **may be considered** as cheating by Vanguard. 

We don't know, let's wait and see.

**Comment by Soraclee** [comment]
Yes I totally understand I use the Riot api whether it's the api or the lcu but it's true that for the injection Vanguard could react strangely, let's wait and see what happens.

**Comment by nomi-san** [comment]
Our method is injecting into Ux, not the Client, it's totally safe.
And in theory, Pengu should work fine when Vanguard is out come to LoL.

Look at Blitz.gg, it does exist a couple years for Valorant even with Vanguard.
Its overlay and injection method are no different to r3nzskin, but valid as a verified 3rd application.

**Comment by Soraclee** [comment]
Yes, but one thing I don't understand is Blitz's claim that it doesn't use any injection, that it doesn't read or write to memory, I rather think it's a lie. As for overlaying, I can't think of anything other than dll injection, because overlaying a window on lol or Valorant wouldn't work properly without dll injection. But r3nzskin could well be blacklisted, as Vanguard doesn't allow skin changes under any circumstances. Let's see how it works and if anything new comes up.

**Comment by Soraclee** [comment]
I'm back to tell you that I've been able to test it and it's all working properly without a hitch, now we'll just have to see what it does in the long term ^^.

**Comment by Kennn-dev** [comment]
Hey i dont know why i got banned , but i just use pengu to develop my plugin and got banned after couple months , this maybe not correctly 100% . but be careful guys  😥
![image](https://github.com/PenguLoader/PenguLoader/assets/59522126/6dc13af7-c775-4809-be87-89873b0ce76e)

**Comment by rumi-chan** [comment]
> Hey i dont know why i got banned , but i just use pengu to develop my plugin and got banned after couple months , this maybe not correctly 100% . but be careful guys 😥 ![image](https://private-user-images.githubusercontent.com/59522126/332447114-6dc13af7-c775-4809-be87-89873b0ce76e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYyOTk3ODcsIm5iZiI6MTcxNjI5OTQ4NywicGF0aCI6Ii81OTUyMjEyNi8zMzI0NDcxMTQtNmRjMTNhZjctYzc3NS00ODA5LWJlODctODk4NzNiMGNlNzZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIxVDEzNTEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlZDA4ZjNlYjFhMmRmY2YxN2VmNWY2YWE4NTljNzRhN2RhNTI0OTUwMGE0NGJiYTJiMmZiMzdkYzM0OWU4NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ANoubOSTUtyqfE0XcJWIAupiZ5DOWaccYxahjgdteWs)

do u use KbotExt or any lobby reveal plugin?

**Comment by nomi-san** [comment]
@Kennn-dev

If you have using lobby reveal, dont worry its could be 2 weeks ban.
> https://x.com/RiotK3o/status/1792706933952311352

Pengu is safe, but be careful before touching exploit things like lobby reveal.

**Comment by Kennn-dev** [comment]
> @Kennn-dev
> 
> If you have using lobby reveal, dont worry it should be 2 weeks ban.
> 
> > https://x.com/RiotK3o/status/1792706933952311352
> 
> Pengu is safe, but be careful before touching exploit things like lobby reveal.

actual nope , you can check my source [repo](https://github.com/Kennn-dev/aram-pengu-loader-plugin)

 its all about Match auto accept and auto pick champion 😥

**Comment by nomi-san** [comment]
Try to remember, any plugins or tools else?
You can DM me via discord or chat in the server for help.
Anyway please open an unban ticket.

**Comment by Kennn-dev** [comment]
I opened, hope they will unban :(  . i'll DM you if need any help . thankss <3
